client/post-editor/editor-sharing/publicize-options.jsx

```diff
@@ -103,18 +108,15 @@ const EditorSharingPublicizeOptions = React.createClass( {
 	},
 
 	onModuleConnectionPopupClosed: function() {
-		if ( ! this.props.site || ! this.props.site.jetpack ) {
+		const { isJetpack, isPublicizeEnabled, siteId } = this.props;
+		if ( ! isJetpack || ! isPublicizeEnabled ) {
```

Comment:
I have a feeling this was previously in a callback because we couldn't be certain at the time of the popup being closed that the module was instantly active. Would be good to test if this regresses the behavior of enabling the module on a Jetpack site.